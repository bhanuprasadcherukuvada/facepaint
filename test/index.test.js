import React from 'react'
import renderer from 'react-test-renderer'
import { css, sheet, flush } from 'emotion'

import facepaint from '../src/index'

const mq = facepaint([
  '@media(min-width: 420px)',
  '@media(min-width: 920px)',
  '@media(min-width: 1120px)'
])

describe('facepaint', () => {
  afterEach(() => flush())
  test('basic', () => {
    const result = css(mq({ color: ['red', 'green', 'blue', 'darkorchid'] }))
    expect(result).toMatchSnapshot()
    const tree = renderer.create(<div css={result}>Basic</div>).toJSON()
    expect(tree).toMatchSnapshot()
    expect(sheet).toMatchSnapshot()
  })

  test('holes', () => {
    const result = css(mq({ color: ['red', , 'blue', 'darkorchid'] }))
    expect(result).toMatchSnapshot()
    const tree = renderer.create(<div css={result}>Basic</div>).toJSON()
    expect(tree).toMatchSnapshot()
    expect(sheet).toMatchSnapshot()
  })

  test('multiple', () => {
    const result = css(
      mq({
        color: ['red', 'green', 'blue', 'darkorchid'],
        display: ['flex', 'block', 'inline-block', 'table'],
        fontSize: 12,
        alignItems: 'center'
      })
    )
    expect(result).toMatchSnapshot()
    const tree = renderer.create(<div css={result}>multiple</div>).toJSON()
    expect(tree).toMatchSnapshot()
    expect(sheet).toMatchSnapshot()
  })

  test('nested', () => {
    const result = css(
      mq({
        backgroundColor: 'hotpink',
        textAlign: 'center',
        width: ['25%', '50%', '75%', '100%'],
        '& .foo': {
          color: ['red', 'green', 'blue', 'darkorchid'],
          '& img': {
            height: [10, 15, 20, 25]
          }
        }
      })
    )
    expect(result).toMatchSnapshot()
    const tree = renderer
      .create(
        <div css={result}>
          <div className="foo">foo</div>
          function
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
    expect(sheet).toMatchSnapshot()
  })
})
