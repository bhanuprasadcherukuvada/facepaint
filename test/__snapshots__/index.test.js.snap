// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`facepaint array values with selectors 1`] = `
.glamor-0 .current-index {
  margin-right: 15px;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
}

.glamor-0 .current-index {
  color: blue;
}

.glamor-0 .current-index {
  display: none;
}

@media (min-width:420px) {
  .glamor-0 .current-index {
    color: red;
  }
}

@media (min-width:420px) {
  .glamor-0 .current-index {
    display: block;
  }
}

<div
  className="glamor-0"
>
  <div
    className="foo"
  >
    foo
  </div>
  function
</div>
`;

exports[`facepaint array values with selectors 2`] = `
".css-1n491te .current-index {
  margin-right: 15px;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
}

.css-1n491te .current-index {
  color: blue;
}

@media (min-width:420px) {
  .css-1n491te .current-index {
    color: red;
  }
}

.css-1n491te .current-index {
  display: none;
}

@media (min-width:420px) {
  .css-1n491te .current-index {
    display: block;
  }
}"
`;

exports[`facepaint basic 1`] = `
.glamor-0 {
  color: red;
}

@media (min-width:420px) {
  .glamor-0 {
    color: green;
  }
}

@media (min-width:920px) {
  .glamor-0 {
    color: blue;
  }
}

@media (min-width:1120px) {
  .glamor-0 {
    color: darkorchid;
  }
}

<div
  className="glamor-0"
>
  Basic
</div>
`;

exports[`facepaint basic 2`] = `
".css-17umrpw {
  color: red;
}

@media (min-width:420px) {
  .css-17umrpw {
    color: green;
  }
}

@media (min-width:920px) {
  .css-17umrpw {
    color: blue;
  }
}

@media (min-width:1120px) {
  .css-17umrpw {
    color: darkorchid;
  }
}"
`;

exports[`facepaint boolean, null, and undefined values 1`] = `
.glamor-0 {
  color: blue;
  color: red;
}

<div
  className="glamor-0"
>
  <div
    className="foo"
  >
    foo
  </div>
  function
</div>
`;

exports[`facepaint boolean, null, and undefined values 2`] = `
".css-dhb7kq {
  color: blue;
  color: red;
}"
`;

exports[`facepaint holes 1`] = `
.glamor-0 {
  color: red;
}

@media (min-width:920px) {
  .glamor-0 {
    color: blue;
  }
}

@media (min-width:1120px) {
  .glamor-0 {
    color: darkorchid;
  }
}

<div
  className="glamor-0"
>
  Basic
</div>
`;

exports[`facepaint holes 2`] = `
".css-1myvor3 {
  color: red;
}

@media (min-width:920px) {
  .css-1myvor3 {
    color: blue;
  }
}

@media (min-width:1120px) {
  .css-1myvor3 {
    color: darkorchid;
  }
}"
`;

exports[`facepaint multiple 1`] = `
.glamor-0 {
  font-size: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.glamor-0 {
  color: red;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:420px) {
  .glamor-0 {
    color: green;
    display: block;
  }
}

@media (min-width:920px) {
  .glamor-0 {
    color: blue;
    display: inline-block;
  }
}

@media (min-width:1120px) {
  .glamor-0 {
    color: darkorchid;
    display: table;
  }
}

<div
  className="glamor-0"
>
  multiple
</div>
`;

exports[`facepaint multiple 2`] = `
".css-dfpbx9 {
  font-size: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.css-dfpbx9 {
  color: red;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:420px) {
  .css-dfpbx9 {
    color: green;
    display: block;
  }
}

@media (min-width:920px) {
  .css-dfpbx9 {
    color: blue;
    display: inline-block;
  }
}

@media (min-width:1120px) {
  .css-dfpbx9 {
    color: darkorchid;
    display: table;
  }
}"
`;

exports[`facepaint nested 1`] = `
.glamor-0 {
  background-color: hotpink;
  text-align: center;
}

.glamor-0 {
  width: 25%;
}

.glamor-0 .foo {
  color: red;
}

.glamor-0 .foo img {
  height: 10px;
}

@media (min-width:420px) {
  .glamor-0 {
    width: 50%;
  }
}

@media (min-width:920px) {
  .glamor-0 {
    width: 75%;
  }
}

@media (min-width:1120px) {
  .glamor-0 {
    width: 100%;
  }
}

@media (min-width:420px) {
  .glamor-0 .foo {
    color: green;
  }
}

@media (min-width:920px) {
  .glamor-0 .foo {
    color: blue;
  }
}

@media (min-width:1120px) {
  .glamor-0 .foo {
    color: darkorchid;
  }
}

@media (min-width:420px) {
  .glamor-0 .foo img {
    height: 15px;
  }
}

@media (min-width:920px) {
  .glamor-0 .foo img {
    height: 20px;
  }
}

@media (min-width:1120px) {
  .glamor-0 .foo img {
    height: 25px;
  }
}

<div
  className="glamor-0"
>
  <div
    className="foo"
  >
    foo
  </div>
  function
</div>
`;

exports[`facepaint nested 2`] = `
".css-1xa5b6k {
  background-color: hotpink;
  text-align: center;
}

.css-1xa5b6k {
  width: 25%;
}

@media (min-width:420px) {
  .css-1xa5b6k {
    width: 50%;
  }
}

@media (min-width:920px) {
  .css-1xa5b6k {
    width: 75%;
  }
}

@media (min-width:1120px) {
  .css-1xa5b6k {
    width: 100%;
  }
}

.css-1xa5b6k .foo {
  color: red;
}

@media (min-width:420px) {
  .css-1xa5b6k .foo {
    color: green;
  }
}

@media (min-width:920px) {
  .css-1xa5b6k .foo {
    color: blue;
  }
}

@media (min-width:1120px) {
  .css-1xa5b6k .foo {
    color: darkorchid;
  }
}

.css-1xa5b6k .foo img {
  height: 10px;
}

@media (min-width:420px) {
  .css-1xa5b6k .foo img {
    height: 15px;
  }
}

@media (min-width:920px) {
  .css-1xa5b6k .foo img {
    height: 20px;
  }
}

@media (min-width:1120px) {
  .css-1xa5b6k .foo img {
    height: 25px;
  }
}"
`;

exports[`facepaint nested arrays 1`] = `
.glamor-0 {
  color: red;
}

@media (min-width:420px) {
  .glamor-0 {
    color: blue;
  }
}

@media (min-width:920px) {
  .glamor-0 {
    color: darkorchid;
  }
}

<div
  className="glamor-0"
>
  Basic
</div>
`;

exports[`facepaint nested arrays 2`] = `
".css-1kdodwg {
  color: red;
}

@media (min-width:420px) {
  .css-1kdodwg {
    color: blue;
  }
}

@media (min-width:920px) {
  .css-1kdodwg {
    color: darkorchid;
  }
}"
`;

exports[`facepaint pseudo 1`] = `
.glamor-0 {
  background-color: hotpink;
  text-align: center;
}

.glamor-0 {
  width: 25%;
}

.glamor-0:hover {
  width: 50%;
}

.glamor-0:active {
  width: 75%;
}

.glamor-0:focus {
  width: 100%;
}

.glamor-0 .foo {
  color: red;
}

.glamor-0 .foo:hover {
  color: green;
}

.glamor-0 .foo:active {
  color: blue;
}

.glamor-0 .foo:focus {
  color: darkorchid;
}

.glamor-0 .foo img {
  height: 10px;
}

.glamor-0 .foo img:hover {
  height: 15px;
}

.glamor-0 .foo img:active {
  height: 20px;
}

.glamor-0 .foo img:focus {
  height: 25px;
}

<div
  className="glamor-0"
>
  <div
    className="foo"
  >
    foo
  </div>
  function
</div>
`;

exports[`facepaint pseudo 2`] = `
".css-13tn68s {
  background-color: hotpink;
  text-align: center;
}

.css-13tn68s {
  width: 25%;
}

.css-13tn68s:hover {
  width: 50%;
}

.css-13tn68s:active {
  width: 75%;
}

.css-13tn68s:focus {
  width: 100%;
}

.css-13tn68s .foo {
  color: red;
}

.css-13tn68s .foo:hover {
  color: green;
}

.css-13tn68s .foo:active {
  color: blue;
}

.css-13tn68s .foo:focus {
  color: darkorchid;
}

.css-13tn68s .foo img {
  height: 10px;
}

.css-13tn68s .foo img:hover {
  height: 15px;
}

.css-13tn68s .foo img:active {
  height: 20px;
}

.css-13tn68s .foo img:focus {
  height: 25px;
}"
`;

exports[`facepaint repeating 1`] = `
.glamor-0 {
  color: red;
}

@media (min-width:420px) {
  .glamor-0 {
    color: blue;
  }
}

@media (min-width:11200px) {
  .glamor-0 {
    color: darkorchid;
  }
}

<div
  className="glamor-0"
>
  Basic
</div>
`;

exports[`facepaint repeating 2`] = `
".css-1ki7d2p {
  color: red;
}

@media (min-width:420px) {
  .css-1ki7d2p {
    color: blue;
  }
}

@media (min-width:11200px) {
  .css-1ki7d2p {
    color: darkorchid;
  }
}"
`;

exports[`facepaint undefined 1`] = `
.glamor-0 {
  color: red;
}

@media (min-width:920px) {
  .glamor-0 {
    color: blue;
  }
}

@media (min-width:1120px) {
  .glamor-0 {
    color: darkorchid;
  }
}

<div
  className="glamor-0"
>
  Basic
</div>
`;

exports[`facepaint undefined 2`] = `
".css-1myvor3 {
  color: red;
}

@media (min-width:920px) {
  .css-1myvor3 {
    color: blue;
  }
}

@media (min-width:1120px) {
  .css-1myvor3 {
    color: darkorchid;
  }
}"
`;
